x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_creation/helpers.ts

```diff
@@ -420,6 +420,9 @@ export const formatDefineStepData = (defineStepData: DefineStepRule): DefineStep
         query: ruleFields.queryBar?.query?.query as string,
         saved_id: ruleFields.queryBar?.saved_id ?? undefined,
         ...(ruleType === 'threshold' && {
+          ...(ruleFields.enableThresholdSuppression && {
```

Comment:
nit: just think it's clearer to have first he main threshold data, and only after the thresh suppresion data


```suggestion
        ...(ruleType === 'threshold' && {
          threshold: {
            field: ruleFields.threshold?.field ?? [],
            value: parseInt(ruleFields.threshold?.value, 10) ?? 0,
            cardinality:
              !isEmpty(ruleFields.threshold.cardinality?.field) &&
              ruleFields.threshold.cardinality?.value != null
                ? [
                    {
                      field: ruleFields.threshold.cardinality.field[0],
                      value: parseInt(ruleFields.threshold.cardinality.value, 10),
                    },
                  ]
                : [],
          },
          ...(ruleFields.enableThresholdSuppression && {
            alert_suppression: { duration: ruleFields.groupByDuration },
          }),
        }),
```